src/main/java/com/spotify/dns/DnsSrvResolver.java

```diff
@@ -25,10 +26,24 @@
   /**
    * Does a DNS SRV lookup for the supplied fully qualified domain name, and returns the
    * matching results.
+   * @deprecated
+   * This method is deprecated in favor of the asynchronous version.
+   * Use {@link DnsSrvResolver#resolveAsync(String)} instead
    *
    * @param fqdn a DNS name to query for
    * @return a possibly empty list of matching records
    * @throws DnsException if there was an error doing the DNS lookup
    */
+  @Deprecated
   List<LookupResult> resolve(String fqdn);
+
+  /**
+   * Does a DNS SRV lookup for the supplied fully qualified domain name, and returns the
+   * matching results.
+   *
+   * @param fqdn a DNS name to query for
+   * @return a possibly empty list of matching records
+   * @throws DnsException if there was an error doing the DNS lookup
+   */
+  CompletionStage<List<LookupResult>> resolveAsync(String fqdn);
```

Comment:
Just realized that this is a breaking change since we have some internal implementations of this public interface. To make it backward compatible we can do like this I think:

```suggestion
  default CompletionStage<List<LookupResult>> resolveAsync(String fqdn) {
    throw new java.lang.UnsupportedOperationException("Not implemented");
  }
```